mempool/reactor.go

```diff
@@ -131,13 +144,26 @@ func (memR *MempoolReactor) GetChannels() []*p2p.ChannelDescriptor {
 // AddPeer implements Reactor.
 // It starts a broadcast routine ensuring all txs are forwarded to the given peer.
 func (memR *MempoolReactor) AddPeer(peer p2p.Peer) {
-	memR.ids.ReserveForPeer(peer)
+	_ = memR.ids.ReserveForPeer(peer)
 	go memR.broadcastTxRoutine(peer)
 }
 
 // RemovePeer implements Reactor.
 func (memR *MempoolReactor) RemovePeer(peer p2p.Peer, reason interface{}) {
 	memR.ids.Reclaim(peer)
+
+	// XXX (melekes): commented out for performance reasons
+	//
+	// Remove the sender if present to avoid a new node joining and not receiving
+	// a tx because we think it already has it. For that two things should happen:
+	//   1) node has to connect to over 65536 peers during its lifetime
+	//   2) tx must reside in the mempool for the above duration
+	//
+	// for e := memR.Mempool.txs.Front(); e != nil; e = e.Next() {
+	// 	memTx := e.Value.(*mempoolTx)
+	// 	memTx.senders.Delete(id)
+	// }
```

Comment:
So it is not only performance reasons why this is not added but this is also unlikely (that both 1) and 2) are true), right?